app/src/main/java/org/mozilla/focus/activity/ScreenNavigator.java

```diff
@@ -21,12 +32,31 @@
  * 1. Move more methods from MainMediator to ScreenNavigator
  * 2. Finally remove MainMediator from MainActivity
  */
-public class ScreenNavigator {
+public class ScreenNavigator implements LifecycleObserver {
```

Comment:
f you use Java 8 Language, then observe events with DefaultLifecycleObserver. To include it you should add "android.arch.lifecycle:common-java8:<version>" to your build.gradle file. 

If you use Java 7 Language, Lifecycle events are observed using annotations. Once Java 8 Language becomes mainstream on Android, annotations will be deprecated, so between DefaultLifecycleObserver and annotations, you must always prefer DefaultLifecycleObserver.

Reply:
I'll create another PR for this